numpy/__init__.pyi

```diff
@@ -982,6 +983,8 @@ if sys.version_info >= (3, 11):
     _ConvertibleToComplex: TypeAlias = SupportsComplex | SupportsFloat | SupportsIndex | _CharLike_co
 else:
     _ConvertibleToComplex: TypeAlias = complex | SupportsComplex | SupportsFloat | SupportsIndex | _CharLike_co
+_ConvertibleToTD64: TypeAlias = dt.timedelta | int | _CharLike_co | character | number | timedelta64 | np.bool | None
+_ConvertibleToDT64: TypeAlias = dt.date | int | _CharLike_co | character | number | datetime64 | np.bool | None
```

Comment:
How come this has `character` and `_CharLike_co`?  Is that intentional?

Is this missing `dt.datetime`?

Reply:
> I could sneak it in this PR, or open another PR for it?

Not sure I care, if you think it's simple enough (IIUC, this is about the scalar creation?).  FWIW, I would expect them to use the identical definitions to scalar assignment.

W.r.t to the strings, what surprises me was that `_ConvertibleToComplex` does not include `character`.

The other thing is that you have `_ConvertibleToDT64: TypeAlias = dt.date`.  Should it not also have `dt.datetime`?